docs/next/components/mdx/MDXComponents.tsx

```diff
@@ -27,6 +27,38 @@ import DbtModelAssetExplanation from './includes/dagster/integrations/DbtModelAs
 
 export const SearchIndexContext = React.createContext(null);
 
+// https://www.30secondsofcode.org/react/s/use-hash
+// Modified to check for window existence (for nextjs) before accessing
+const useHash = (): string => {
+  const [hash, setHash] = React.useState('');
+
+  const windowExists = typeof window !== 'undefined';
+  useEffect(() => {
+    if (windowExists) {
+      setHash(window.location.hash);
+    }
+  }, [windowExists]);
+
+  const hashChangeHandler = React.useCallback(() => {
+    if (!windowExists) {
+      return;
+    }
+    setHash(window.location.hash);
+  }, [windowExists]);
+
+  React.useEffect(() => {
+    if (!windowExists) {
+      return;
+    }
+    window.addEventListener('hashchange', hashChangeHandler);
+    return () => {
+      window.removeEventListener('hashchange', hashChangeHandler);
+    };
+  }, [windowExists, hashChangeHandler]);
+
+  return hash;
```

Comment:
I don't think we need to be concerned about `window` changing from defined to undefined or vice versa, so would this work?

```
const [hash, setHash] = React.useState(() => {
  return typeof window === 'undefined' ? '' : window.location.hash;
});

useEffect(() => {
  if (typeof window === 'undefined') {
    return;
  }

  const handler = () => {
    setHash(window.location.hash);
  }

  window.addEventListener('hashchange', handler);
  return () => {
    window.removeEventListener('hashchange', handler);
  }
}, []);

return hash;
```